src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -111,14 +112,46 @@ private static void checkHasNaturalAlignment(FunctionDescriptor descriptor) {
     private static void checkHasNaturalAlignmentRecursive(MemoryLayout layout) {
         checkHasNaturalAlignment(layout);
         if (layout instanceof GroupLayout gl) {
+            boolean isCheckRequired = true;
             for (MemoryLayout member : gl.memberLayouts()) {
-                checkHasNaturalAlignmentRecursive(member);
+                isCheckRequired = isAlignmentCheckRequired(member, isCheckRequired);
+                if (isCheckRequired) {
+                    checkHasNaturalAlignmentRecursive(member);
+                }
```

Comment:
This doesn't take into account the recursive native of layouts; for example, the rules for a `double` following a sequence which ends with an `int` should be the same as for a `double` immediately following an `int`.

Reply:
As discussed with @keithc-ca offline, we only need to check the alignment of `double` on AIX which should be an overridden method `hasNaturalAlignment` of `AbstractLayout` in `ValueLayouts.OfDoubleImpl`.

I will need to verify on AIX with the latest changes to ensure it works as expected.